iree/compiler/InputConversion/MHLO/test/scatter.mlir

```diff
@@ -0,0 +1,96 @@
+// RUN: iree-opt -split-input-file -iree-mhlo-to-linalg-on-tensors -canonicalize %s | FileCheck %s
+
+func @still_scatter(%arg0: tensor<1x2x64x12x64xf32>, %arg1: tensor<1x2x4xi32>, %arg2: tensor<1x2x2x1x2x64xf32>) -> tensor<1x2x64x12x64xf32> {
+  %0 = "mhlo.scatter"(%arg0, %arg1, %arg2) ({
+  ^bb0(%arg3: tensor<f32>, %arg4: tensor<f32>):
+    %1 = "mhlo.add"(%arg3, %arg4) : (tensor<f32>, tensor<f32>) -> tensor<f32>
+    "mhlo.return"(%1) : (tensor<f32>) -> ()
+  }) {indices_are_sorted = true, scatter_dimension_numbers = #mhlo.scatter<update_window_dims = [0, 2, 3, 4], inserted_window_dims = [1], scatter_dims_to_operand_dims = [0, 1, 2, 3], index_vector_dim = 2>, unique_indices = true} : (tensor<1x2x64x12x64xf32>, tensor<1x2x4xi32>, tensor<1x2x2x1x2x64xf32>) -> tensor<1x2x64x12x64xf32>
+  return %0 : tensor<1x2x64x12x64xf32>
+}
+
+// CHECK: #[[MAP:.+]] = affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d2, d3, d4)>
+// CHECK:       func @still_scatter
+// CHECK-SAME:    %[[ARG0:[a-zA-Z0-9_]*]]
+// CHECK-SAME:    %[[ARG1:[a-zA-Z0-9_]*]]
+// CHECK-SAME:    %[[ARG2:[a-zA-Z0-9_]*]]
+// CHECK-DAG:     %[[C3:.+]] = arith.constant 3 : index
+// CHECK-DAG:     %[[C1:.+]] = arith.constant 1 : index
+// CHECK-DAG:     %[[C2:.+]] = arith.constant 2 : index
+// CHECK-DAG:     %[[C0:.+]] = arith.constant 0 : index
+// CHECK-DAG:     %[[EXP:.+]] = tensor.expand_shape %[[ARG2]] {{\[\[}}0], [1], [2, 3], [4], [5], [6]]
+// CHECK-DAG:     %[[CARG1:.+]] = tensor.collapse_shape %[[ARG1]] {{\[\[}}0, 1], [2]]
+// CHECK-DAG:     %[[CARG2:.+]] = tensor.collapse_shape %[[EXP]] {{\[\[}}0, 1, 2], [3], [4], [5], [6]]
+// CHECK-DAG:     %[[SCF:.+]] = scf.for %[[ARG3:.+]] = %[[C0]] to %[[C2]] step %[[C1]] iter_args(%[[ARG4:.+]] = %[[ARG0]])
+// CHECK-DAG:       %[[MUL:.+]] = arith.muli %[[ARG3]], %[[C2]]
```

Comment:
I think this is def not the thing we want in IREE. This makes batch dim non-parallel and could make it run on host side.

I think the missing things in linalg_ext.scatter op is that we do not consider `inserted_window_dims`. IIUC, this is still a scatter op but we would like to apply multiplication on indices of `updates`.

A better way is to extend linalg_ext.scatter op to handle the case. It is still tile-able to me. We may make linalg_ext.scatter op carry a bit more information (like multiplier?) and update the generateScalarImplementation method, (i.e., create a mul op before accessing indices).

@rsuderman would you like to take into how to extend linalg_ext op?

Reply:
Mahesh and I had a discussion on how to support `mhlo.scatter` via `linalg_ext` and agreed that trying to fully support `mhlo` semantics is not a good idea. The premise that any dimension (indexed or not) could have any slice size makes it very difficult to support the operation.

For this case, we need to synchronize after each scatter to avoid scattering to overlapping regions. When this becomes overly inefficient we can address the related performance issues.